src/NodeServices/Attributes.cs

```diff
@@ -289,21 +290,59 @@ public PreferredShortNameAttribute(string preferredShortName)
     /// <summary> 
     /// This attribute indicates the node is obsolete
     /// </summary> 
-    [AttributeUsage(AttributeTargets.Method)] 
-    public class IsObsoleteAttribute : Attribute 
-    { 
-        public string Message { get; protected set; } 
- 
-        public IsObsoleteAttribute() 
-        { 
-            Message = String.Empty; 
-        } 
- 
-        public IsObsoleteAttribute(string message) 
-        { 
-            Message = message; 
-        } 
-    } 
+    [AttributeUsage(AttributeTargets.Method| AttributeTargets.Constructor | AttributeTargets.Property)]
+    public class IsObsoleteAttribute : Attribute
+    {
+        public string Message { get; protected set; }
+
+        public IsObsoleteAttribute()
+        {
+            Message = String.Empty;
+        }
+
+        public IsObsoleteAttribute(string message)
+        {
+            Message = message;
+        }
+
+        /// <summary>
+        /// Attribute constructor which enables localized message lookup.
+        /// </summary>
+        /// <param name="descriptionResourceID">resx id for this resource</param>
+        /// <param name="resourceType">type that contains resource strings.</param>
+        /// <exception cref="ArgumentNullException"></exception>
+        public IsObsoleteAttribute(string descriptionResourceID, Type resourceType)
+        {
+            LookupResourceByID(descriptionResourceID, resourceType);
+        }
+        /// <summary>
+        /// Attribute constructor which enables localized message lookup.
+        /// </summary>
+        /// <param name="descriptionResourceID">resx id for this resource</param>
+        /// <param name="typeName">name of type that contains resource strings.</param>
+        /// <exception cref="ArgumentNullException"></exception>
+        public IsObsoleteAttribute(string descriptionResourceID, string typeName)
```

Comment:
this is very useful for autogen because we can't reference the type inside the resource assembly in some situations - but we know the type name and assembly name.